packages/headless/src/controllers/core/search-parameter-manager/headless-core-search-parameter-manager.ts

```diff
@@ -145,6 +142,31 @@ export function enrichParameters(
   };
 }
 
+/**
+ * This function is essential to ensure that the `tab` parameter is valid.
+ * We need it to be valid prior to dispatching the `restoreSearchParameters` action since the facet logic relies on the `tab` parameter to determine which facets to show.
+ * If the `tab` parameter is invalid, it can lead to unexpected behavior in the facets.
+ */
+export function ensureTabIsValid(
+  tabSet: TabSetState | undefined,
+  parameters: SearchParameters
+): SearchParameters {
+  if (parameters.tab && tabSet) {
+    const tabExists = Object.values(tabSet).some(
+      (tab) => tab.id === parameters.tab
+    );
+    const currentActiveTab = Object.values(tabSet).find((tab) => tab.isActive);
+
+    if (!tabExists && currentActiveTab) {
+      parameters.tab = currentActiveTab.id;
```

Comment:
This function isn't pure: it mutates a property in the received `parameters` object. 

Preferably, we'd work on a copy of the object.